### PR TITLE
Add glob feature

### PR DESCRIPTION
**Pull request by FWirtz**
- [x] Add directory support for dst.
- [x] Add Create directory if none exists.
- [x] Add missing import errno to utils.
- [x] Deal with faulty paths (i.e. Skip)
- [x] Global src/dst handling
- [x] Add to_images
- [x] Add to_dcim
- [x] Test

### CODE REVIEW

File: bin/medconv [hunk 1]
```diff
@@ -19,20 +21,25 @@ class MedConv(object):
...
         self._dst = dst
+        self._foldername = None
+        if len(self._src) > 1 and self._dst is not None:
+            self._foldername = self._dst
 
-        if not os.path.exists(self._src) and not os.path.isdir(self._src):
-            print('The give src path does not exist: {}'.format(self._src))
+        for path in self._src:
+            if not os.path.exists(path) and not os.path.isdir(path):
+                print('The given src path does not exist: {}'.format(path))
```
Comment (mischkew) [inline bin/medconv]: if a path does not exist, either:
- exit the program
- skip the path and continue exectuting normally. if no path matches exist the program

Comment (FWirtz) [inline bin/medconv]: Hm, like the 2nd idea better.
Skip path and once finished show a list of paths that failed?

Comment (FWirtz) [inline bin/medconv]: Well thats BS: glob only returns existing paths, so if a path is wrong it is simply skipped & not forwarded to the program. Therefore, 2nd idea and no information that a path matches.
Instead: `X files were succesfully converted.`

**Comment by FWirtz** [comment]
## Src/Dst Handling
```
if (dst) AND (src==1):
    -> Output  filename supplied
if (dst) AND (src > 1):
    -> Output Foldername supplied
else:
    -> Nothing supplied, input=output

+= extension
```

**Comment by FWirtz** [comment]
@adius @simsieg I get a `Sorry, can't find the code check you are looking for` when clicking on the `Details` above.
Can you have a look?

**Comment by FWirtz** [comment]
@mischkew I think this is good to go.
I am however not quite sure how to test / what the expected behaviour is.
`to_video` works as I expect it to.
Can you have a look at the others & review?

**Comment by mischkew** [comment]
Feram is beta. So you can just ignore their status check.

**Comment by adius** [comment]
Haha, dammit. Same error as last time. It's your branch name: `feature/#2-glob-inputs` the `#` is destroying our url. I guess we need to properly url-encode it. 😅
Thanks for uncovering this one!

**Comment by adius** [comment]
If you encode the `#` it works: https://app.feram.io//#/codechecks/ThinkSono~medconv-feature~%232-glob-inputs-8246b531d4a9b6abe159363497d2b1976c6f9f90

**Comment by FWirtz** [comment]
@adius I see!
Great suggestions by Feram, thanks!
Didn't see the typos.
Really like the .dcm change-requests haha, might wanna disable fix-suggestions for bytecode. 😊

File: bin/medconv [hunk 2]
```diff
@@ -14,98 +16,101 @@ from medconv.conversions import \
...
-            print('The give src path does not exist: {}'.format(self._src))
-
-    def _guess_conversion(self):
-        type = Conversion.guess_type(self._src)
+    def _get_output_filename(self, path, extension):
+        if self._dst is not None and len(self._src) == 1:
+            # single filename supplied
+            filename = self._dst
+        else:
+            __, basename = os.path.split(path)
```
Comment (mischkew) [inline bin/medconv]: this works just fine. when not using the path/ dirname though, you can just use `os.path.basename` function.

File: bin/medconv [hunk 3]
```diff
@@ -14,98 +16,101 @@ from medconv.conversions import \
...
+            name, __ = os.path.splitext(basename)
+            if self._dst is not None:
+                # folder supplied
+                folder = os.path.abspath(os.path.expanduser(self._dst))
+                utils.make_sure_path_exists(folder)
+                filename = folder + "/" + name
```
Comment (mischkew) [inline bin/medconv]: use `os.path.join` for this. also let's try to stick to single quotes wherever possible.

File: bin/medconv [hunk 4]
```diff
@@ -14,98 +16,101 @@ from medconv.conversions import \
...
         return {
             Conversion.TYPES['DIRECTORY']: ImagesConversion,
             Conversion.TYPES['NIFTI']: NiftiConversion,
             Conversion.TYPES['VIDEO']: VideoConversion,
             Conversion.TYPES['DICOM']: DicomConversion
-        }[type](self._src)
+        }[conversion_type](path)
 
     def to_images(self, output_format='png', prefix='image'):
         """Convert DICOM, Video or Nifti files into a sequence of images.
```
Comment (mischkew) [inline bin/medconv]: when converting files into images, the directory is created with a file extension, e.g. `my-data.dcm` --> `my-data.png/`. it should not append the image extension to the directory name.

**Comment by adius** [comment]
@FWirtz
- Encoding URLs is merged
- Not checking dcm files is in the works: https://github.com/sindresorhus/binary-extensions/pull/13

**Comment by adius** [comment]
@FWirtz Why didn't you let Feram commit the typo fixes if you liked them?

**Comment by FWirtz** [comment]
@adius sgtm!
I wanted to! But once I clicked on commit on Feram it told me that it couldn't find the code I was looking for, so I thought that it might be due to the encoding problem. 🙈

**Comment by adius** [comment]
@FWirtz Ok, thanks for the feedback. Then I guess the link was also broken internally. If you push another commit onto this branch feram should be able to pick it up and provide committable suggestions.

**Comment by FWirtz** [comment]
Thanks, will do, probably tomorrow.
Will also delete some comments here to clean up, hope thats fine.

**Comment by adius** [comment]
👍, sure!

File: bin/medconv (hunk 4 above)
Comment (FWirtz) [inline bin/medconv]: You are absolutely right, I did not notice that.
Fixed in next commit.

**Comment by FWirtz** [comment]
Hey @adius,
I still cannot let Feram apply the fixes. Once I click on `Submit` Feram replies with "Sorry, I cannot find the code check you are looking for". 😢

**Comment by FWirtz** [comment]
@mischkew Ready for merge. Approve?

**Pull request merged**